### PR TITLE
Align subscription_mailer and user_mailer from behavior

### LINKED ISSUE #590: host takes presedence over from in SMTP config, breaking relay
__Steps to reproduce__:

1. Configure host in `config/settings.yml`:

    ```yaml
    :host: "subdomain.domain.com"
    ```

2. Configure smtp in `config/settings.yml`:

    ```yaml
    :smtp:
      :address: "smtp.sparkpostmail.com"
      :port: 587
      :from: "no-reply@domain.com"
      :enable_starttls_auto: true
      :authentication: :plain
      :user_name: "SMTP_Injection"
      :password: "<password>"
    ```
3. Add Note under different user accounts to trigger email

__What happens__: email that is relayed is constructed with `host` setting, e.g. 

```
From: "<user>" <no-reply@subdomain.domain.com>
```

causing DKIM check for verified senders to fail on SparkPost (or any other external service).

__What should happen__: Email should be relayed as by SMTP configuration, with 

```
From: "<user>" <no-reply@domain.com>
```

### PR DESCRIPTION
**Pull request by johnnyshields**
Fixes #590
Closes #591

All mailers should be both using the smtp from address defined.

Also:
- adds #dig method to Settings (to mirror Hash)
- removes duplicate spec run of mailers (it was being run by travis twice)

### CODE REVIEW

File: spec/mailers/subscription_mailer_spec.rb [hunk 1]
```diff
@@ -34,16 +26,21 @@
...
+    it "uses email defined in settings as the sender" do
+      allow(Setting).to receive(:dig).with(:email_comment_replies, :address).and_return("email_comment_reply@example.com")
+      expect(mail.from).to eql(["email_comment_reply@example.com"])
     end
 
-    it "should set default reply-to address if email and host don't exist" do
-      allow(Setting.email_comment_replies).to receive(:[]).with(:address).and_return("")
-      allow(Setting).to receive(:host).and_return("")
-      expect(mail.from).to eql(["no-reply@example.com"])
+    it "should fallback to smtp from addres as the sender" do
```
Comment (CloCkWeRX) [inline spec/mailers/subscription_mailer_spec.rb]: Spelling

File: .travis.yml [hunk 2]
```diff
@@ -39,7 +39,6 @@ before_script:
 # Run specs
 script: 
   - "RAILS_ENV=test bundle exec rake spec:models"
-  - "RAILS_ENV=test bundle exec rake spec:mailers"
```
Comment (CloCkWeRX) [inline .travis.yml]: Is that intentional? We still have spec/mailers/subscription_mailer_spec.rb

Comment (CloCkWeRX) [reply .travis.yml]: Ah, its a duplicate, nevermind

**Comment by johnnyshields** [comment]
@CloCkWeRX your change to test runs in Travis is making the whole thing take far longer than previous. I'd suggest to roll it back and also only test Ruby 2.4 (with 2.3 as an allowed failure)

**Pull request merged**